### ISSUE TITLE
Feature Request: Sampler Device

### ISSUE THREAD

**Issue by Jurek-Raben** [bitwig/dawproject#67]
Hi,

please consider to add a generic Sampler device, which supports:

- Sample layers, per layer start/end note, offset
- Per sample loop, start point, end point, loop mode (foward/ ping pong)
- Volume ADSR

Thanks

**Comment by git-moss** [comment]
What would be the advantage of this? Both the Multisample format (except ADSR) and SFZ can do this and are open documented.

**Comment by Jurek-Raben** [comment]
My request is about covering the standard set of a daw in daw project. A bunch of daws even do not support streamed audio clips, but using a (re-/ "real")sampler instead. Also most daws use a sampler in some kind for drums, e.g. Bitwig. These cases could be covered with a generic sampler device. Maybe "type=drums/tonal/atonal" what be a good attribute, too. I mean just the mininal featureset of a re-sampler-device (being about to change pitch on-the-fly).

I would even consider a generic sampler device as more important than a generic compressor or limiter, since these drastically can be different and its hard to achieve an unique sound.

**Comment by tapir** [comment]
I was surprised to see that the DAW format supported any generic devices at all. I don't think it's wise to have it in the first place because even a simple device like compressor can have multitude of different parameters and the resulting audio will for sure not be bit-perfect from DAW to DAW.

However if this was the decision made and we have to live with it than I would argue that a generic sampler device is much more important than a compressor or gate. First of all, with a minimal amount of parameters (as stated in the original post) you can almost guarantee an exact result in any DAW but also it's a very fundamental part of producing. 

SFZ is definitely a no go as creating a SFZ instrument is cumbersome and I don't see anybody choosing it over a simple sampler if you're not working with 10-20+ samples.

Long story short, +1 for this feature.

**Comment by RustoMCSpit** [comment]
> I was surprised to see that the DAW format supported any generic devices at all. I don't think it's wise to have it in the first place because even a simple device like compressor can have multitude of different parameters and the resulting audio will for sure not be bit-perfect from DAW to DAW.

well the only way to work around that is separate daw agnostic plugins. unless you want bitwig to shares its dsp with everyone and vice versa. it's the closest thing you can get
